cli/js/file_info.ts

```diff
@@ -62,7 +78,16 @@ export class FileInfoImpl implements FileInfo {
     // Only non-null if on Unix
     this.dev = isUnix ? dev : null;
     this.ino = isUnix ? ino : null;
-    this.mode = isUnix ? mode : null;
+    this.mode = isUnix ? mode & 0o7777 : null;
+    this.type = isUnix
+      ? mode & ~0o7777
+      : res.isFile
+      ? FileType.TYPE_REGULAR
+      : res.isDir
+      ? FileType.TYPE_DIRECTORY
+      : res.isSymlink
+      ? FileType.TYPE_SYMLINK
+      : null;
```

Comment:
In POSIX and Go these flags are left in the same field (st_mode). I think we ought to just do the same.

Reply:
Ok, how about I evolve this PR to suppress all the parts except what I called (a) before (don't treat the three options as being exclusive).

Reply:
I'm not sure what that means. Those methods, `isDirectory()`, etc are in the Go API - I think they should stay. Their existence doesn't imply they are exhaustive.

Reply:
cli/js/file_info.ts now has:

```
  isDirectory(): boolean {
    return !this.#isFile && !this.#isSymlink;
  }
```

which is wrong --- it counts fifos etc as directories. And to fix this, the Rust side has to provide more info in the StatResponse.

Reply:
We need the StatResponse to also have a isDirectory boolean, and then to stash that as a private var on the FileInfo too.

Reply:
Indeed, that is wrong. Your change, adding isDir to the StatResponse is better.

I just don't agree with the further change of adding this new "type" member.

Reply:
Ok, I'll push more commits to this PR Monday evening that roll back most of the changes besides those. Then I have to rework the rest of my queue to cohere with that. But once a later form of this PR is merged, I should be able to push 5 more substantive PRs (adding create/createNew in more places). After that, the highlights are: the fstat etc ops, broader nofollow functionality (as in lstat vs stat), and openat etc. functions. So many of the patches are so interdependent, they have to keep coming in steps.